src/main/java/org/apache/commons/text/TextStringBuilder.java

```diff
@@ -1850,7 +1850,25 @@ public boolean equals(final Object obj) {
      * @return true if the builders contain the same characters in the same order
      */
     public boolean equals(final TextStringBuilder other) {
-        return other != null && Arrays.equals(buffer, other.buffer);
+        if (this == other) {
+            return true;
+        }
+        if (other == null) {
+            return false;
+        }
+        if (this.size != other.size) {
+            return false;
+        }
+        // Be aware not to use Arrays.equals(buffer, other.buffer) for equals() method
+        // as length of the buffers may be different (TEXT-211)
```

Comment:
They cannot be different by the time you get here though, can they?

Reply:
size of the TextStringBuilder cannot be different but sizes of their buffers can as the size represents the real number of characters of the TextStringBuilder but the buffer may have been initialized with another capacity.